cmd/arktos-network-controller/app/controller.go

```diff
@@ -158,6 +158,33 @@ func (c *Controller) process(item interface{}) {
 	c.queue.Forget(item)
 }
 
+// manageNonFlatNetwork is the core logic to manage a non-flat typed network object
+func manageNonFlatNetwork(net *v1.Network, netClient arktos.Interface, svcClient kubernetes.Interface, toDeployDNS bool, domainName string) error {
+	var svc *corev1.Service
+	var err error
+	if len(net.Status.DNSServiceIP) == 0 {
```

Comment:
I think there is a logic error. If len(net.Status.DNSServiceIP) >0, line 166 will not be run, so svc will be nil, then 177 will hit nil pointer error.

It seems to me the logic here should be the same as line 163-177 in func manageFlatNetwork(). It that is the case, let's have a small helper func for those code.

Reply:
good point; need to protect from nil svc object.
for the code reuse, as the logic seems quite different from the flat typed network, it is fine to have separate code body. Having said that; if in the future the code structure is changed and code reuse makes more sense, we should refactor code accordingly.